netmaster/mastercfg/networkstate.go

```diff
@@ -43,19 +43,24 @@ const (
 // vlans with ovs. The state is stored as Json objects.
 type CfgNetworkState struct {
 	core.CommonState
-	Tenant      string        `json:"tenant"`
-	NetworkName string        `json:"networkName"`
-	NwType      string        `json:"nwType"`
-	PktTagType  string        `json:"pktTagType"`
-	PktTag      int           `json:"pktTag"`
-	ExtPktTag   int           `json:"extPktTag"`
-	SubnetIP    string        `json:"subnetIP"`
-	SubnetLen   uint          `json:"subnetLen"`
-	Gateway     string        `json:"gateway"`
-	EpAddrCount int           `json:"epAddrCount"`
-	EpCount     int           `json:"epCount"`
-	IPAllocMap  bitset.BitSet `json:"ipAllocMap"`
-	DNSServer   string        `json:"dnsServer"`
+	Tenant        string          `json:"tenant"`
+	NetworkName   string          `json:"networkName"`
+	NwType        string          `json:"nwType"`
+	PktTagType    string          `json:"pktTagType"`
+	PktTag        int             `json:"pktTag"`
+	ExtPktTag     int             `json:"extPktTag"`
+	SubnetIP      string          `json:"subnetIP"`
+	SubnetLen     uint            `json:"subnetLen"`
+	Gateway       string          `json:"gateway"`
+	EpAddrCount   int             `json:"epAddrCount"`
+	EpCount       int             `json:"epCount"`
+	IPAllocMap    bitset.BitSet   `json:"ipAllocMap"`
+	DNSServer     string          `json:"dnsServer"`
+	IPv6Subnet    string          `json:"ipv6SubnetIP"`
+	IPv6SubnetLen uint            `json:"ipv6SubnetLen"`
+	IPv6Gateway   string          `json:"ipv6Gateway"`
+	IPv6AllocMap  map[string]bool `json:"ipv6AllocMap"`
```

Comment:
The last allocated host-id is saved and we allocate the next one (last+1) if it is available. When the number of containers is close to max-hosts and containers allocate/release addresses, there will be a walk to find the available address.